src/Jackett.Common/Models/IndexerConfig/ConfigurationDataBasicLoginOrApiKey.cs

```diff
@@ -0,0 +1,25 @@
+namespace Jackett.Common.Models.IndexerConfig
+{
+    public class ConfigurationDataBasicLoginOrAPIKey : ConfigurationData
+    {
+        public DisplayItem Instructions { get; private set; }
+        public bool UseAPIKey { get; private set; }
```

Comment:
Question: What happens if you declare all variables: apikey, username, password, 2fa... but you only initialize those which are used? Are they displayed in the configuration dialog?
It will be nice to avoid `AddDynamic` because the code is more clean and you ger auto-complete in `configData.`

Reply:
Any variables that are declared as properties and are of type `Item` will be included in the configuration dialog no matter if it's initialized or otherwise.

The only way I can see this done without using `AddDynamic` is one of the following approaches:

1. Modify `GetItems` in `ConfigurationData` to include non-null value properties only. Right now, all properties are included regardless if it's initialized.
2. Add an extra variable in `Item` which enables the option of including it in the JSON response or otherwise.

I wanted to avoid touching `ConfigurationData`, but I have decided to go with option 1, and exclude `null` properties. That way, if it's not initialized, it will not be included.

Reply:
Good!